tests/impl.cpp

```diff
@@ -948,134 +263,159 @@ bool test_mm_set_epi8(const int8_t *_a)
                         d12, d13, d14, d15);
 }
 
-bool test_mm_set1_epi32(int32_t i)
+result_t test_mm_set1_epi32(const SSE2NEONTestImpl &impl, uint32_t i)
 {
-    __m128i a = _mm_set1_epi32(i);
-    return validateInt32(a, i, i, i, i);
+    int32_t x = impl.mTestInts[i];
+    __m128i a = _mm_set1_epi32(x);
+    return validateInt32(a, x, x, x, x);
 }
 
-bool testret_mm_set_epi32(int32_t x, int32_t y, int32_t z, int32_t w)
+result_t test_mm_set_epi32(const SSE2NEONTestImpl &impl, uint32_t i)
 {
+    int32_t x = impl.mTestInts[0];
+    int32_t y = impl.mTestInts[1];
+    int32_t z = impl.mTestInts[2];
+    int32_t w = impl.mTestInts[3];
     __m128i a = _mm_set_epi32(x, y, z, w);
     return validateInt32(a, w, z, y, x);
 }
 
-__m128i test_mm_set_epi32(int32_t x, int32_t y, int32_t z, int32_t w)
+__m128i do_mm_set_epi32(int32_t x, int32_t y, int32_t z, int32_t w)
 {
     __m128i a = _mm_set_epi32(x, y, z, w);
     validateInt32(a, w, z, y, x);
     return a;
 }
 
-bool test_mm_store_ps(float *p, float x, float y, float z, float w)
+result_t do_mm_store_ps(float *p, float x, float y, float z, float w)
 {
     __m128 a = _mm_set_ps(x, y, z, w);
     _mm_store_ps(p, a);
     ASSERT_RETURN(p[0] == w);
     ASSERT_RETURN(p[1] == z);
     ASSERT_RETURN(p[2] == y);
     ASSERT_RETURN(p[3] == x);
-    return true;
+    return TEST_SUCCESS;
 }
 
-bool test_mm_store_ps(int32_t *p, int32_t x, int32_t y, int32_t z, int32_t w)
+result_t do_mm_store_ps(int32_t *p, int32_t x, int32_t y, int32_t z, int32_t w)
```

Comment:
These tests are not modified because of?

Reply:
That was my fault :b. 
My intention to change some `test_xxx` functions to `do_xxx` functions is because the return value of these functions are not `result_t`, which should not be executed as test cases. (e.g. `__m128 do_mm_load_ps`). For these functions, we should another `test_xxx` functions that returns `result_t`

Reply:
Apparently, these two functions should keep the `test` prefix rather than the `do` prefix.